Submitted By:            DJ Lucas <dj_AT_linuxfromscratch_DOT_org>
Date:                    2016-12-27
Initial Package Version: 3.12.4
Upstream Status:         Not applicable
Origin:                  Self, rediffed for nss-3.28.
Description:             Adds auto-generated nss.pc and nss-config script, and
                         allows building without nspr in the source tree.
                         For 3.35, Requires: updated to nspr >= 4.18.

diff -Naurp nss-3.28-orig/nss/Makefile nss-3.28/nss/Makefile
--- nss-3.28-orig/nss/Makefile	2016-12-21 05:56:27.000000000 -0600
+++ nss-3.28/nss/Makefile	2016-12-26 22:24:52.695146032 -0600
@@ -46,7 +46,7 @@ include $(CORE_DEPTH)/coreconf/rules.mk
 # (7) Execute "local" rules. (OPTIONAL).                              #
 #######################################################################
 
-nss_build_all: build_nspr all latest
+nss_build_all: all latest
 
 nss_clean_all: clobber_nspr clobber
 
diff -Naurp nss-3.28-orig/nss/config/Makefile nss-3.28/nss/config/Makefile
--- nss-3.28-orig/nss/config/Makefile	1969-12-31 18:00:00.000000000 -0600
+++ nss-3.28/nss/config/Makefile	2016-12-26 22:20:40.008205774 -0600
@@ -0,0 +1,40 @@
+CORE_DEPTH = ..
+DEPTH      = ..
+
+include $(CORE_DEPTH)/coreconf/config.mk
+
+NSS_MAJOR_VERSION = `grep "NSS_VMAJOR" ../lib/nss/nss.h | awk '{print $$3}'`
+NSS_MINOR_VERSION = `grep "NSS_VMINOR" ../lib/nss/nss.h | awk '{print $$3}'`
+NSS_PATCH_VERSION = `grep "NSS_VPATCH" ../lib/nss/nss.h | awk '{print $$3}'`
+PREFIX = /usr
+
+all: export libs
+
+export:
+	# Create the nss.pc file
+	mkdir -p $(DIST)/lib/pkgconfig
+	sed -e "s,@prefix@,$(PREFIX)," \
+	    -e "s,@exec_prefix@,\$${prefix}," \
+	    -e "s,@libdir@,\$${prefix}/lib," \
+	    -e "s,@includedir@,\$${prefix}/include/nss," \
+	    -e "s,@NSS_MAJOR_VERSION@,$(NSS_MAJOR_VERSION),g" \
+	    -e "s,@NSS_MINOR_VERSION@,$(NSS_MINOR_VERSION)," \
+	    -e "s,@NSS_PATCH_VERSION@,$(NSS_PATCH_VERSION)," \
+	    nss.pc.in > nss.pc
+	chmod 0644 nss.pc
+	ln -sf ../../../../nss/config/nss.pc $(DIST)/lib/pkgconfig
+
+	# Create the nss-config script
+	mkdir -p $(DIST)/bin
+	sed -e "s,@prefix@,$(PREFIX)," \
+	    -e "s,@NSS_MAJOR_VERSION@,$(NSS_MAJOR_VERSION)," \
+	    -e "s,@NSS_MINOR_VERSION@,$(NSS_MINOR_VERSION)," \
+	    -e "s,@NSS_PATCH_VERSION@,$(NSS_PATCH_VERSION)," \
+	    nss-config.in > nss-config
+	chmod 0755 nss-config
+	ln -sf ../../../nss/config/nss-config $(DIST)/bin
+
+libs:
+
+dummy: all export libs
+
diff -Naurp nss-3.28-orig/nss/config/nss-config.in nss-3.28/nss/config/nss-config.in
--- nss-3.28-orig/nss/config/nss-config.in	1969-12-31 18:00:00.000000000 -0600
+++ nss-3.28/nss/config/nss-config.in	2016-12-26 22:20:40.008205774 -0600
@@ -0,0 +1,153 @@
+#!/bin/sh
+
+prefix=@prefix@
+
+major_version=@NSS_MAJOR_VERSION@
+minor_version=@NSS_MINOR_VERSION@
+patch_version=@NSS_PATCH_VERSION@
+
+usage()
+{
+	cat <<EOF
+Usage: nss-config [OPTIONS] [LIBRARIES]
+Options:
+	[--prefix[=DIR]]
+	[--exec-prefix[=DIR]]
+	[--includedir[=DIR]]
+	[--libdir[=DIR]]
+	[--version]
+	[--libs]
+	[--cflags]
+Dynamic Libraries:
+	nss
+	nssutil
+	smime
+	ssl
+	softokn
+EOF
+	exit $1
+}
+
+if test $# -eq 0; then
+	usage 1 1>&2
+fi
+
+lib_nss=yes
+lib_nssutil=yes
+lib_smime=yes
+lib_ssl=yes
+lib_softokn=yes
+
+while test $# -gt 0; do
+  case "$1" in
+  -*=*) optarg=`echo "$1" | sed 's/[-_a-zA-Z0-9]*=//'` ;;
+  *) optarg= ;;
+  esac
+
+  case $1 in
+    --prefix=*)
+      prefix=$optarg
+      ;;
+    --prefix)
+      echo_prefix=yes
+      ;;
+    --exec-prefix=*)
+      exec_prefix=$optarg
+      ;;
+    --exec-prefix)
+      echo_exec_prefix=yes
+      ;;
+    --includedir=*)
+      includedir=$optarg
+      ;;
+    --includedir)
+      echo_includedir=yes
+      ;;
+    --libdir=*)
+      libdir=$optarg
+      ;;
+    --libdir)
+      echo_libdir=yes
+      ;;
+    --version)
+      echo ${major_version}.${minor_version}.${patch_version}
+      ;;
+    --cflags)
+      echo_cflags=yes
+      ;;
+    --libs)
+      echo_libs=yes
+      ;;
+    nss)
+      lib_nss=yes
+      ;;
+    nssutil)
+      lib_nssutil=yes
+      ;;
+    smime)
+      lib_smime=yes
+      ;;
+    ssl)
+      lib_ssl=yes
+      ;;
+    softokn)
+      lib_softokn=yes
+      ;;
+    *)
+      usage 1 1>&2
+      ;;
+  esac
+  shift
+done
+
+# Set variables that may be dependent upon other variables
+if test -z "$exec_prefix"; then
+    exec_prefix=`pkg-config --variable=exec_prefix nss`
+fi
+if test -z "$includedir"; then
+    includedir=`pkg-config --variable=includedir nss`
+fi
+if test -z "$libdir"; then
+    libdir=`pkg-config --variable=libdir nss`
+fi
+
+if test "$echo_prefix" = "yes"; then
+    echo $prefix
+fi
+
+if test "$echo_exec_prefix" = "yes"; then
+    echo $exec_prefix
+fi
+
+if test "$echo_includedir" = "yes"; then
+    echo $includedir
+fi
+
+if test "$echo_libdir" = "yes"; then
+    echo $libdir
+fi
+
+if test "$echo_cflags" = "yes"; then
+    echo -I$includedir
+fi
+
+if test "$echo_libs" = "yes"; then
+      libdirs="-L$libdir"
+      if test -n "$lib_nss"; then
+	libdirs="$libdirs -lnss${major_version}"
+      fi
+      if test -n "$lib_nssutil"; then
+        libdirs="$libdirs -lnssutil${major_version}"
+      fi
+      if test -n "$lib_smime"; then
+	libdirs="$libdirs -lsmime${major_version}"
+      fi
+      if test -n "$lib_ssl"; then
+	libdirs="$libdirs -lssl${major_version}"
+      fi
+      if test -n "$lib_softokn"; then
+        libdirs="$libdirs -lsoftokn${major_version}"
+      fi
+      echo $libdirs
+fi      
+
diff -Naurp nss-3.28-orig/nss/config/nss.pc.in nss-3.28/nss/config/nss.pc.in
--- nss-3.28-orig/nss/config/nss.pc.in	1969-12-31 18:00:00.000000000 -0600
+++ nss-3.28/nss/config/nss.pc.in	2016-12-26 22:22:53.300694346 -0600
@@ -0,0 +1,12 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: NSS
+Description: Network Security Services
+Version: @NSS_MAJOR_VERSION@.@NSS_MINOR_VERSION@.@NSS_PATCH_VERSION@
+Requires: nspr >= 4.18
+Libs: -L@libdir@ -lnss@NSS_MAJOR_VERSION@ -lnssutil@NSS_MAJOR_VERSION@ -lsmime@NSS_MAJOR_VERSION@ -lssl@NSS_MAJOR_VERSION@ -lsoftokn@NSS_MAJOR_VERSION@
+Cflags: -I${includedir}
+
diff -Naurp nss-3.28-orig/nss/manifest.mn nss-3.28/nss/manifest.mn
--- nss-3.28-orig/nss/manifest.mn	2016-12-21 05:56:27.000000000 -0600
+++ nss-3.28/nss/manifest.mn	2016-12-26 22:24:12.278991843 -0600
@@ -10,4 +10,4 @@ IMPORTS =	nspr20/v4.8 \
 
 RELEASE = nss
 
-DIRS = coreconf lib cmd cpputil gtests
+DIRS = coreconf lib cmd cpputil gtests config
diff '--exclude=.deps' '--exclude=*.log' -ur nss-3.38.orig/nss/lib/freebl/blake2b.h nss-3.38/nss/lib/freebl/blake2b.h
--- nss-3.38.orig/nss/lib/freebl/blake2b.h	2018-08-05 18:40:10.629684000 -0700
+++ nss-3.38/nss/lib/freebl/blake2b.h	2018-08-05 19:05:53.179899000 -0700
@@ -9,7 +9,10 @@
 #define BLAKE_H
 
 #include <stddef.h>
-#include <stdint.h>
+/* #include <stdint.h> */
+
+#define UINT64_MAX (18446744073709551615ULL)
+
 
 struct Blake2bContextStr {
     uint64_t h[8];                     /* chained state */
diff '--exclude=.deps' '--exclude=*.log' -ur nss-3.38.orig/nss/lib/freebl/ecl/curve25519_32.c nss-3.38/nss/lib/freebl/ecl/curve25519_32.c
--- nss-3.38.orig/nss/lib/freebl/ecl/curve25519_32.c	2018-08-05 18:40:10.639670000 -0700
+++ nss-3.38/nss/lib/freebl/ecl/curve25519_32.c	2018-08-05 19:05:53.179932000 -0700
@@ -9,7 +9,7 @@
 #include "ecl-priv.h"
 #include "mpi.h"
 
-#include <stdint.h>
+/* #include <stdint.h> */
 #include <stdio.h>
 
 typedef uint32_t elem[32];
diff '--exclude=.deps' '--exclude=*.log' -ur nss-3.38.orig/nss/lib/freebl/gcm.h nss-3.38/nss/lib/freebl/gcm.h
--- nss-3.38.orig/nss/lib/freebl/gcm.h	2018-08-05 18:40:10.719557000 -0700
+++ nss-3.38/nss/lib/freebl/gcm.h	2018-08-05 19:05:53.179962000 -0700
@@ -6,7 +6,7 @@
 #define GCM_H 1
 
 #include "blapii.h"
-#include <stdint.h>
+/* #include <stdint.h> */
 
 #ifdef NSS_X86_OR_X64
 /* GCC <= 4.8 doesn't support including emmintrin.h without enabling SSE2 */
diff '--exclude=.deps' '--exclude=*.log' -ur nss-3.38.orig/nss/lib/freebl/rijndael.h nss-3.38/nss/lib/freebl/rijndael.h
--- nss-3.38.orig/nss/lib/freebl/rijndael.h	2018-08-05 18:40:10.749540000 -0700
+++ nss-3.38/nss/lib/freebl/rijndael.h	2018-08-05 19:05:53.179977000 -0700
@@ -6,7 +6,7 @@
 #define _RIJNDAEL_H_ 1
 
 #include "blapii.h"
-#include <stdint.h>
+/* #include <stdint.h> */
 
 #if defined(NSS_X86_OR_X64)
 /* GCC <= 4.8 doesn't support including emmintrin.h without enabling SSE2 */
diff '--exclude=.deps' '--exclude=*.log' -ur nss-3.38.orig/nss/lib/freebl/verified/kremlib.h nss-3.38/nss/lib/freebl/verified/kremlib.h
--- nss-3.38.orig/nss/lib/freebl/verified/kremlib.h	2018-08-05 18:40:10.759593000 -0700
+++ nss-3.38/nss/lib/freebl/verified/kremlib.h	2018-08-05 19:21:04.970021000 -0700
@@ -206,6 +206,21 @@
 /* ... for Solaris */
 #elif defined(__sun__)
 #include <sys/byteorder.h>
+
+#define AT_U64(x) ((uint64_t)(x))
+
+#define SWAPLS32(x) ((uint64_t)ntohl((uint32_t)((x) & 0xffffffff)))
+
+#define SWAP64(x) ((SWAPLS32(AT_U64(x)) << 32) | (SWAPLS32(AT_U64(x) >> 32))) 
+
+#if defined(_BIG_ENDIAN)
+#define BE_64(x) (x)
+#define LE_64(x) (SWAP64(x))
+#else
+#define LE_64(x) (x)
+#define BE_64(x) (SWAP64(x))
+#endif
+
 #define htole64(x) LE_64(x)
 #define le64toh(x) LE_64(x)
 #define htobe64(x) BE_64(x)
diff '--exclude=.deps' '--exclude=*.log' -ur nss-3.38.orig/nss/lib/util/secport.c nss-3.38/nss/lib/util/secport.c
--- nss-3.38.orig/nss/lib/util/secport.c	2018-08-05 18:40:11.219622000 -0700
+++ nss-3.38/nss/lib/util/secport.c	2018-08-05 19:05:53.189510000 -0700
@@ -21,7 +21,7 @@
 #include "prenv.h"
 #include "prinit.h"
 
-#include <stdint.h>
+/* #include <stdint.h> */
 
 #ifdef DEBUG
 #define THREADMARK
diff '--exclude=.deps' '--exclude=*.log' -ur nss-3.38.orig/nss/lib/util/secport.h nss-3.38/nss/lib/util/secport.h
--- nss-3.38.orig/nss/lib/util/secport.h	2018-08-05 18:40:11.219627000 -0700
+++ nss-3.38/nss/lib/util/secport.h	2018-08-05 19:05:53.189526000 -0700
@@ -45,7 +45,7 @@
 #include <string.h>
 #include <stddef.h>
 #include <stdlib.h>
-#include <stdint.h>
+/* #include <stdint.h> */
 #include "prtypes.h"
 #include "prlog.h" /* for PR_ASSERT */
 #include "plarena.h"
